client/web/src/components/readonly-toggle.tsx

```diff
@@ -0,0 +1,142 @@
+import cx from "classnames"
+import React, { useCallback, useState } from "react"
+import { AuthResponse, AuthType } from "src/hooks/auth"
+import { NodeData } from "src/hooks/node-data"
+import { ReactComponent as ChevronDown } from "src/icons/chevron-down.svg"
+import { ReactComponent as Eye } from "src/icons/eye.svg"
+import { ReactComponent as User } from "src/icons/user.svg"
+import Popover from "src/ui/popover"
+import ProfilePic from "src/ui/profile-pic"
+
+export default function ReadonlyToggle({
+  node,
+  auth,
+  newSession,
+}: {
+  node: NodeData
+  auth: AuthResponse
+  newSession: () => Promise<void>
+}) {
+  const [open, setOpen] = useState<boolean>(false)
+
+  return (
+    <Popover
+      className="p-3 bg-white rounded-lg shadow flex flex-col gap-2 max-w-[317px]"
+      content={
+        <ReadonlyPopoverContent
+          node={node}
+          auth={auth}
+          newSession={newSession}
+        />
+      }
+      side="bottom"
+      align="end"
+      open={open}
+      onOpenChange={setOpen}
+      asChild
+    >
+      {!auth.canManageNode ? (
+        <button
+          className={cx(
+            "pl-3 py-1 bg-zinc-800 rounded-full flex justify-start items-center",
+            { "pr-1": auth.sourceIdentity, "pr-3": !auth.sourceIdentity }
+          )}
+          onClick={() => setOpen(!open)}
+        >
+          <Eye />
+          <div className="text-white leading-snug ml-2 mr-1">Viewing</div>
+          <ChevronDown className="stroke-white w-[15px] h-[15px]" />
+          {auth.sourceIdentity && (
+            <ProfilePic
+              className="ml-2"
+              size="medium"
+              url={auth.sourceIdentity.profilePicUrl}
+            />
+          )}
+        </button>
+      ) : (
+        <div
+          className={cx(
+            "w-[34px] h-[34px] p-1 rounded-full items-center inline-flex",
+            {
+              "bg-transparent": !open,
+              "bg-neutral-300": open,
+            }
+          )}
+        >
+          <button onClick={() => setOpen(!open)}>
+            <ProfilePic
+              size="medium"
+              url={auth.sourceIdentity?.profilePicUrl}
+            />
+          </button>
+        </div>
+      )}
+    </Popover>
+  )
+}
+
+function ReadonlyPopoverContent({
+  node,
+  auth,
+  newSession,
+}: {
+  node: NodeData
+  auth: AuthResponse
+  newSession: () => Promise<void>
+}) {
+  const handleSignInClick = useCallback(() => {
+    if (window.location.hostname != node.IP) {
+      window.location.href = `http://${node.IP}:5252/?check=now`
+    } else {
+      newSession()
+    }
+  }, [node.IP, window.location.href, newSession])
+
+  return (
+    <>
+      <div className="text-black text-sm font-medium leading-tight">
+        {!auth.canManageNode ? "Viewing" : "Managing"}
+        {auth.sourceIdentity && ` as ${auth.sourceIdentity.loginName}`}
+      </div>
+      {!auth.canManageNode &&
```

Comment:
I think the grouping of these values is getting evaluated incorrectly.  It's doing:

```
if (!canManage && (!auth.sourceIdentity || authNeeded == tailscale) ) {
  // login button
} else {
  // You don't have permission
}
```

Whereas what we want is:

```
if (!canManage) {
  if (!auth.sourceIdentity || authNeeded == tailscale) ) {
    // login button
  } else {
    // You don't have permission
  }
}
```

Reply:
Good catch, thank you!